### PR TITLE
docs: add workflows for docs

### PR DESCRIPTION
**Pull request by renancaraujo**
## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [x] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by scarletteliza — APPROVED** [review]

**Review by scarletteliza — APPROVED** [review]

**Review by alestiago — APPROVED** [review]

**Review by felangel — COMMENTED** [review]
We should also update https://github.com/VeryGoodOpenSource/very_good_workflows/blob/3d53e081c0d710e53b09041a36163a59e10f436b/site/docusaurus.config.js#LL11C2-L11C34 and all other references of my-site.

**Comment by scarletteliza** [comment]
> We should also update https://github.com/VeryGoodOpenSource/very_good_workflows/blob/3d53e081c0d710e53b09041a36163a59e10f436b/site/docusaurus.config.js#LL11C2-L11C34 and all other references of my-site.

do we have to prior to merging or nah?

**Comment by felangel** [comment]
> > We should also update https://github.com/VeryGoodOpenSource/very_good_workflows/blob/3d53e081c0d710e53b09041a36163a59e10f436b/site/docusaurus.config.js#LL11C2-L11C34 and all other references of my-site.
> 
> do we have to prior to merging or nah?

I think you should otherwise you'll deploy a site with broken links.

**Comment by scarletteliza** [comment]
https://github.com/VeryGoodOpenSource/very_good_workflows/pull/121

**Review by scarletteliza — DISMISSED** [review]

File: site/docusaurus.config.js [hunk 1]
```diff
@@ -37,7 +37,8 @@ const config = {
           sidebarPath: require.resolve('./sidebars.js'),
           // Please change this to your repo.
           // Remove this to remove the "edit this page" links.
-          editUrl: 'https://github.com/my-org/my_docs_site/tree/main/',
+          editUrl:
+            'https://github.com/VeryGoodOpenSource/ very_good_workflows/tree/main/',
```
Comment (scarletteliza) [inline site/docusaurus.config.js]:
```suggestion
          editUrl:
            'https://github.com/VeryGoodOpenSource/very_good_workflows/tree/main/site',
```

File: .github/workflows/site_deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: deploy
+
+on:
+  push:
+    paths:
+      - ".github/workflows/site_deploy.yaml"
+      - "deploy/**"
```
Comment (felangel) [inline .github/workflows/site_deploy.yaml:7]:
```suggestion
      - "site/**"
```

File: site/docusaurus.config.js [hunk 3]
```diff
@@ -37,7 +37,8 @@ const config = {
           sidebarPath: require.resolve('./sidebars.js'),
           // Please change this to your repo.
           // Remove this to remove the "edit this page" links.
-          editUrl: 'https://github.com/my-org/my_docs_site/tree/main/',
+          editUrl:
+            'https://github.com/VeryGoodOpenSource/very_good_workflows/tree/main/',
```
Comment (scarletteliza) [inline site/docusaurus.config.js:41]:
```suggestion
          editUrl:
            'https://github.com/VeryGoodOpenSource/very_good_workflows/tree/main/site',
```

Comment (felangel) [inline site/docusaurus.config.js:41]:
```suggestion
            'https://github.com/VeryGoodOpenSource/very_good_workflows/tree/main/',
```
```suggestion
            'https://github.com/VeryGoodOpenSource/very_good_workflows/tree/main/site',
```

**Review by felangel — APPROVED** [review]

File: site/docusaurus.config.js (hunk 3 above)
Comment (renancaraujo) [reply site/docusaurus.config.js]: yall finish each others sentences https://github.com/VeryGoodOpenSource/very_good_workflows/pull/120#discussion_r1226940789

Comment (scarletteliza) [reply site/docusaurus.config.js]: 😆

**Pull request merged**